### PR TITLE
Move GPU LUT files into the tests dir

### LINKED ISSUE #2017: Improve Lut3D GPU unit test to not create temporary files
The module: OpenColorIO/tests/gpu/Lut3DOp_test.cpp, uses a function createTempFile to execute some tests with various test LUTs. These are not later cleaned up. These LUTs could be added to the test file directory and read from there so that no temp files are created. The createTempFile function could then be removed.

### PR DESCRIPTION
**Pull request by ananthb**
Closes #2017.

Instead of creating temporary LUT files for GPU tests, store them in source code instead.
This removes the need to clean them up after each run.

### CODE REVIEW

**Review by doug-walker — CHANGES_REQUESTED** [review]
Thanks for the contribution Ananth!

Apologies for the delayed review, it's been very busy here.

There are a few minor things I'd like you to change before we merge it.

File: tests/cpu/Context_tests.cpp
Comment (ananthb) [reply tests/cpu/Context_tests.cpp:4]: The changes are on line 121 and 145 to remove references to the now deleted `GPUHelpers.h` file.

My editor is setup to run clang-format automatically, but I can see how this could be annoying.
I'll revert the whitespace changes.

File: tests/gpu/Lut3DOp_test.cpp
Comment (doug-walker) [inline tests/gpu/Lut3DOp_test.cpp:138]: Thanks for the updates Ananth!  However, it looks like you have a duplicate copy of the tests below that use the earlier LUT names. Please remove the duplicate tests.

Ideally, you should run the unit tests locally before pushing updates.   :)

**Review by doug-walker — CHANGES_REQUESTED** [review]

File: tests/gpu/Lut3DOp_test.cpp
Comment (ananthb) [reply tests/gpu/Lut3DOp_test.cpp]: I knew that would come back to bite me. Done.

File: tests/cpu/Context_tests.cpp
Comment (doug-walker) [inline tests/cpu/Context_tests.cpp:123]: You're missing a "t".  "GPUUnitTest.h".   Please ensure your tests are passing locally before you push.

**Review by doug-walker — CHANGES_REQUESTED** [review]

File: tests/cpu/Context_tests.cpp
Comment (ananthb) [reply tests/cpu/Context_tests.cpp]: I only ran the gpu tests. Sorry about that. Fixed now.

I'm getting two failures in tests I did not change in the CPU tests suite. Is that normal?

Comment (doug-walker) [reply tests/cpu/Context_tests.cpp]: No, all the CPU tests should pass (they do on the CI machines). You could try to fix them or log them as issues.

**Review by doug-walker — APPROVED** [review]
Thanks Ananth, approving this.  As soon as we have a second approval we will be able to merge.

File: tests/gpu/GPUUnitTest.cpp
Comment (remia) [inline tests/gpu/GPUUnitTest.cpp:8]: Minor: why do we remove iomanip here? Seems like this file uses std::setprecision.

**Review by remia — APPROVED** [review]

File: tests/gpu/GPUUnitTest.cpp
Comment (ananthb) [reply tests/gpu/GPUUnitTest.cpp]: That was my editor being overzealous with auto format again. I reverted the whitespace changes but missed this one. Reverted this as well.

**Pull request merged**
